src/app/clusters/fixed-label-server/fixed-label-server.cpp

```diff
@@ -0,0 +1,112 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/****************************************************************************
+ * @file
+ * @brief Implementation for the Fixed Label Server Cluster
+ ***************************************************************************/
+
+#include <app-common/zap-generated/af-structs.h>
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app-common/zap-generated/ids/Clusters.h>
+#include <app/AttributeAccessInterface.h>
+#include <app/util/attribute-storage.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/logging/CHIPLogging.h>
+#include <platform/PlatformManager.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::FixedLabel;
+using namespace chip::app::Clusters::FixedLabel::Attributes;
+
+namespace {
+
+class DescriptorAttrAccess : public AttributeAccessInterface
+{
+public:
+    // Register for the Fixed Label cluster on all endpoints.
+    DescriptorAttrAccess() : AttributeAccessInterface(Optional<EndpointId>::Missing(), FixedLabel::Id) {}
+
+    CHIP_ERROR Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder) override;
+
+private:
+    static constexpr uint16_t ClusterRevision = 1;
+
+    CHIP_ERROR ReadLabelList(EndpointId endpoint, AttributeValueEncoder & aEncoder);
+    CHIP_ERROR ReadClusterRevision(EndpointId endpoint, AttributeValueEncoder & aEncoder);
+};
+
+constexpr uint16_t DescriptorAttrAccess::ClusterRevision;
+
+CHIP_ERROR DescriptorAttrAccess::ReadLabelList(EndpointId endpoint, AttributeValueEncoder & aEncoder)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    DeviceLayer::LabelList<DeviceLayer::kMaxFixedLabels> labelList;
+
+    if (DeviceLayer::PlatformMgr().GetFixedLabelList(labelList) == CHIP_NO_ERROR)
+    {
+        err = aEncoder.EncodeList([&labelList](const auto & encoder) -> CHIP_ERROR {
+            for (auto label : labelList)
```

Comment:
```suggestion
            for (auto & label : labelList)
```

Reply:
use "auto &" or "const auto &" always give following error on Darwin, but not on other platforms

`../../src/app/clusters/fixed-label-server/fixed-label-server.cpp:61:13: error: cannot bind non-const lvalue reference of type ‘chip::app::Clusters::FixedLabel::Structs::LabelStruct::Type&’ to an rvalue of type ‘chip::app::Clusters::FixedLabel::Structs::LabelStruct::Type’`